ls1intum/Artemis
# `Development`: Refactor competency related domain objects

### Checklist
#### General


#### Server

### Motivation and Context


As the integration of learning paths will add new classes to our domain, we should create a new package to reduce clutter in the domain package.

### Description

This PR moves all competency-related domain Objects into the new package `competency`. Imports are adjusted accordingly.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Students
- 1 Course with at least 1 Competency that existed prior to this PR

Instructor:
1. Navigate to Competency Management
2. Make sure you can see your old Competency
3. Create a new Competency and check that the Competency is created
4. Import a Competency into your course
5. Edit the three Competencies and make sure the changes are saved as expected

Student:
1. Navigate to the Competencies of your course
2. Make sure you can see all Competencies and the correct information is displayed

### Review Progress






#### Code Review
#### Manual Tests

### Test Coverage







#### Server

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [Competency.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5029-1/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.domain.competency/Competency.html) | 78% |  |
| [CompetencyProgress.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5029-1/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.domain.competency/CompetencyProgress.html) | 76% |  |
| [CompetencyRelation.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5029-1/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.domain.competency/CompetencyRelation.html) | 100% |  |
| [CompetencyTaxonomy.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5029-1/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.domain.competency/CompetencyTaxonomy.html) | 100% |  |
| [Course.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5029-1/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.domain/Course.html) | 91% |  |
| [Exercise.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5029-1/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.domain/Exercise.html) | 91% |  |
| [User.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5029-1/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.domain/User.html) | 89% |  |
| [ExerciseUnit.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5029-1/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.domain.lecture/ExerciseUnit.html) | 100% |  |
| [LectureUnit.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5029-1/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.domain.lecture/LectureUnit.html) | 96% |  |
| [CompetencyRepository.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5029-1/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.repository/CompetencyRepository.html) | 47% |  |
| [CompetencyProgressService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5029-1/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service/CompetencyProgressService.html) | 90% |  |
| [CompetencyService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5029-1/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service/CompetencyService.html) | 99% |  |
| [CourseService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5029-1/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service/CourseService.html) | 90% |  |
| [LectureUnitService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5029-1/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service/LectureUnitService.html) | 99% |  |
| [CompetencyResource.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5029-1/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.web.rest/CompetencyResource.html) | 97% |  |
| [PageUtil.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5029-1/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.web.rest.util/PageUtil.html) | 94% |  |
